p2p/simulations/network.go

```diff
@@ -78,41 +78,22 @@ func (self *Network) Events() *event.Feed {
 	return &self.events
 }
 
-// NewNode adds a new node to the network with a random ID
-func (self *Network) NewNode() (*Node, error) {
-	conf := adapters.RandomNodeConfig()
-	conf.Services = []string{self.DefaultService}
-	return self.NewNodeWithConfig(conf)
-}
-
 // NewNodeWithConfig adds a new node to the network with the given config,
 // returning an error if a node with the same ID or name already exists
 func (self *Network) NewNodeWithConfig(conf *adapters.NodeConfig) (*Node, error) {
 	self.lock.Lock()
 	defer self.lock.Unlock()
 
-	// create a random ID and PrivateKey if not set
```

Comment:
The caller is supposed to provide a config for the `NewNode`, but we instead patch it, in case of missing fields. This is happening at a few different levels, which I find very confusing. I'd rather we have one place that we are explicit about the config we want the node to be created with.

Reply:
go ahead and change it i think this is a good idea

Reply:
yes, already changed most of it.